media/js/base/fxa-utm-referral.js

```diff
@@ -178,9 +178,13 @@ if (typeof window.Mozilla === 'undefined') {
         for (var i = 0; i < acceptedParams.length; i++) {
             var acceptedParam = acceptedParams[i];
             if (Object.prototype.hasOwnProperty.call(params, acceptedParam)) {
-                var foundParam = decodeURIComponent(params[acceptedParam]);
-                if (allowedChars.test(foundParam)) {
-                    finalParams[acceptedParam] = foundParam;
+                try {
+                    var foundParam = decodeURIComponent(params[acceptedParam]);
+                    if (allowedChars.test(foundParam)) {
+                        finalParams[acceptedParam] = foundParam;
+                    }
+                } catch (e) {
+                    // silently drop malformed parameter values (issue #10897)
```

Comment:
is it worth still logging the error?

`uns_footer%` seems to be an unsubscribe footer tag that just got mixed in the email/newsletter template syntax. It feels useful to be able to alert other teams that this is failing.(But that's also tied into whether or not anyone is actually looking at the data for these tags)

Reply:
My hesitation here is that these params are ultimately out of our control and can come from many different sources (including ones outside of our organisation), so I'm not really sure how actionable throwing an error here would be. We'd need to 1.) notice the error, and then 2.) figure out where it comes from and 3.) try and notify the owner.

Maybe we can figure out a system for the above in the future, but right now I think avoiding the errors (which halts the rest of the data script from running) is probably the most user-impactful thing to fix.